buildsystem/docgen.cmake

```diff
@@ -0,0 +1,88 @@
+# Copyright 2019-2019 the openage authors. See copying.md for legal info.
+
+# Implement the target 'webdoc' - documentation generation with doxygen and sphinx
+# for the purpose to host online
+
+# Set the output path
+set(WEBDOC_PATH "${CMAKE_BINARY_DIR}/webdoc")
+#set(WEBDOC_OUTPUT_PATH "${CMAKE_CURRENT_SOURCE_DIR}/webdoc-hosting")
+set(WEBDOC_OUTPUT_PATH "${CMAKE_CURRENT_SOURCE_DIR}/../openage-docs/")
+
+# branch 'webdoc build'
+
+if(${SPHINX_FOUND})
+    add_custom_target(webdoc-doxygen
```

Comment:
These targets are per stage right now for debugging, will change that as the last step.